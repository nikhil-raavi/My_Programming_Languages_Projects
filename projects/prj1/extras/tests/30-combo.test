var single22: number;

var
single33
:
string #some comment
#another comment
  ;

var nested_23 : record
  triangle: record  #3 points
    point1: record x: number; y: number; end;
    point2: record x: number; y: number; end;
    point3: record x: number; y: number; end;
  end;
  color: record #RGB components
    r: number; g: number; b: number;
  end;
  index: number;
  pos: record x: number; y: number; end;
end;