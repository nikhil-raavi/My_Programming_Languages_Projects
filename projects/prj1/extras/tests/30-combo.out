[
   [
      "single22",
      "number"
   ],
   [
      "single33",
      "string"
   ],
   [
      "nested_23",
      [
         [
            "triangle",
            [
               [
                  "point1",
                  [
                     [
                        "x",
                        "number"
                     ],
                     [
                        "y",
                        "number"
                     ]
                  ]
               ],
               [
                  "point2",
                  [
                     [
                        "x",
                        "number"
                     ],
                     [
                        "y",
                        "number"
                     ]
                  ]
               ],
               [
                  "point3",
                  [
                     [
                        "x",
                        "number"
                     ],
                     [
                        "y",
                        "number"
                     ]
                  ]
               ]
            ]
         ],
         [
            "color",
            [
               [
                  "r",
                  "number"
               ],
               [
                  "g",
                  "number"
               ],
               [
                  "b",
                  "number"
               ]
            ]
         ],
         [
            "index",
            "number"
         ],
         [
            "pos",
            [
               [
                  "x",
                  "number"
               ],
               [
                  "y",
                  "number"
               ]
            ]
         ]
      ]
   ]
]
